src/test-run/execute-js-expression.js

```diff
@@ -26,7 +42,22 @@ export function executeJsExpression (expression, skipVisibilityCheck, testRun) {
         }
     };
 
-    var context = createContext(sandbox);
+    return createContext(sandbox);
+}
+
+function getExecutionContext (testRun, skipVisibilityCheck) {
+    var contextId = testRun.id;
+
+    if (!testRun || !contexts[testRun.id])
+        contexts[contextId] = createExecutionContext(testRun);
+
+    setContextOption(contextId, 'skipVisibilityCheck', skipVisibilityCheck);
+
+    return contexts[contextId];
+}
+
+export function executeJsExpression (expression, skipVisibilityCheck, testRun) {
```

Comment:
may be replace `expression, skipVisibilityCheck, testRun` -> `expression, testRun, skipVisibilityCheck`, it's better because 
1) option `skipVisibilityCheck` become optional
2) more common with CommandBase constructor, less chance to use wrong order